src/redux/globalSearch/reducer.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Reducer for globalSearch
+ */
+import * as Actions from './actions';
+
+const initialState = {
+  locationSearch: {
+    data: [],
+    query: '',
+    isFetching: false,
+    isFetched: false,
+  },
+};
+
+// the location page reducer
+function locationSearch(state = initialState.locationSearch, action = {}) {
+  switch (action.type) {
+    case Actions.FETCH_LOCATION_SEARCH:
+      return {
+        ...state,
+        data: state.data,
```

Comment:
On second thought, having ...state there is not good since it includes the `error` key.